scijava-ops-image/src/main/java/org/scijava/ops/image/types/ExtendedRAITypeExtractor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * #%L
+ * Image processing operations for SciJava Ops.
+ * %%
+ * Copyright (C) 2014 - 2024 SciJava developers.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package org.scijava.ops.image.types;
+
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.util.Util;
+import net.imglib2.view.ExtendedRandomAccessibleInterval;
+import org.scijava.priority.Priority;
+import org.scijava.types.extract.SubTypeExtractor;
+import org.scijava.types.extract.TypeExtractor;
+import org.scijava.types.extract.TypeReifier;
+
+import java.lang.reflect.Type;
+
+/**
+ * {@link TypeExtractor} plugin which operates on
+ * {@link RandomAccessibleInterval} objects.
+ * <p>
+ * Note that this {@link TypeExtractor} is low priority since we prefer subclass
+ * TypeExtractors to take pre
+ * </p>
+ *
+ * @author Gabriel Selzer
+ */
+public class ExtendedRAITypeExtractor extends
+	SubTypeExtractor<ExtendedRandomAccessibleInterval<?, ?>>
+{
+
+	@Override
+	public double priority() {
+		return Priority.HIGH;
+	}
+
+	@Override
+	public Class<?> baseClass() {
+		return ExtendedRandomAccessibleInterval.class;
+	}
+
+	@Override
+	protected Type[] getTypeParameters(TypeReifier r,
+		ExtendedRandomAccessibleInterval<?, ?> object)
+	{
+		return new Type[] { r.reify(Util.getTypeFromInterval(object.getSource())), r.reify(object.getSource()) };
```

Comment:
The need for this may be obsolete soon, when we finish updating to ImgLib2 v7, which has a dedicated `getType()` method now.

Reply:
This is true, however we still cannot reify `ExtendedRandomAccessibleInterval` without a dedicated type extractor due to its second type variable.

Reply:
At the end of the day, do we *really* need to know about that second type variable? Probably not. But the work has already been done, so why not...